ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/common/validation/Validation.kt

```diff
@@ -11,6 +12,21 @@ data class ValidationContext(
     val violations: MutableSet<ConstraintViolation> = mutableSetOf(),
 )
 
+fun <T> validateParams(fn: ValidationContext.() -> T) = with(ValidationContext()) {
+    try {
+        fn()
+    } catch (e: ConstraintViolationException) {
+        // Partial validation will throw an exception
+        throw OcpiClientInvalidParametersException(e.toReadableString())
```

Comment:
ensure we provide the correct response depending on where validation fails. When executing validation for incoming values, we automatically convert to ocpi client error. All other validations of outgoing will still be constraint violations, which will be correctly turned into ocpi server error.